### PR TITLE
[SPARK-18088][ML] Various ChiSqSelector cleanups

### PR DESCRIPTION
**Pull request by jkbradley**
## What changes were proposed in this pull request?
- Renamed kbest to numTopFeatures
- Renamed alpha to fpr
- Added missing Since annotations
- Doc cleanups
## How was this patch tested?

Added new standardized unit tests for spark.ml.
Improved existing unit test coverage a bit.

### CODE REVIEW

**Comment by jkbradley** [comment]
CC: @mpjlu @srowen @yanboliang

**Comment by jkbradley** [comment]
test this please

**Comment by jkbradley** [comment]
I'll fix these later tonight

File: docs/ml-features.md [hunk 1]
```diff
@@ -1333,14 +1333,14 @@ for more details on the API.
 `ChiSqSelector` stands for Chi-Squared feature selection. It operates on labeled data with
 categorical features. ChiSqSelector uses the
 [Chi-Squared test of independence](https://en.wikipedia.org/wiki/Chi-squared_test) to decide which
-features to choose. It supports three selection methods: `KBest`, `Percentile` and `FPR`:
+features to choose. It supports three selection methods: `numTopFeatures`, `percentile`, `fpr`:
 
-* `KBest` chooses the `k` top features according to a chi-squared test. This is akin to yielding the features with the most predictive power.
-* `Percentile` is similar to `KBest` but chooses a fraction of all features instead of a fixed number.
-* `FPR` chooses all features whose false positive rate meets some threshold.
+* `numTopFeatures` chooses the `k` top features according to a chi-squared test. This is akin to yielding the features with the most predictive power.
```
Comment (mpjlu) [inline docs/ml-features.md]: Should use k here since KBest is changed to numTopFeatures?

File: docs/ml-features.md [hunk 2]
```diff
@@ -1333,14 +1333,14 @@ for more details on the API.
...
+* `percentile` is similar to `numTopFeatures` but chooses a fraction of all features instead of a fixed number.
+* `fpr` chooses all features whose p-value is below a threshold, thus controlling the false positive rate of selection.
 
-By default, the selection method is `KBest`, the default number of top features is 50. User can use
-`setNumTopFeatures`, `setPercentile` and `setAlpha` to set different selection methods.
+By default, the selection method is `numTopFeatures`, with the default number of top features set to 50. User can use
+`setNumTopFeatures`, `setPercentile` and `setFpr` to set different selection methods.
```
Comment (mpjlu) [inline docs/ml-features.md]: setNumTopFeatures, setPercentile, setFpr just set parameter, setSelectorType is used to set selection methods.

File: docs/mllib-feature-extraction.md [hunk 3]
```diff
@@ -227,22 +227,19 @@ both speed and statistical learning behavior.
 [`ChiSqSelector`](api/scala/index.html#org.apache.spark.mllib.feature.ChiSqSelector) implements
 Chi-Squared feature selection. It operates on labeled data with categorical features. ChiSqSelector uses the
 [Chi-Squared test of independence](https://en.wikipedia.org/wiki/Chi-squared_test) to decide which
-features to choose. It supports three selection methods: `KBest`, `Percentile` and `FPR`:
+features to choose. It supports three selection methods: `numTopFeatures`, `percentile`, `fpr`:
 
-* `KBest` chooses the `k` top features according to a chi-squared test. This is akin to yielding the features with the most predictive power.
-* `Percentile` is similar to `KBest` but chooses a fraction of all features instead of a fixed number.
-* `FPR` chooses all features whose false positive rate meets some threshold.
+* `numTopFeatures` chooses the `k` top features according to a chi-squared test. This is akin to yielding the features with the most predictive power.
```
Comment (mpjlu) [inline docs/mllib-feature-extraction.md]: ditto

File: docs/mllib-feature-extraction.md [hunk 4]
```diff
@@ -227,22 +227,19 @@ both speed and statistical learning behavior.
...
+* `percentile` is similar to `numTopFeatures` but chooses a fraction of all features instead of a fixed number.
+* `fpr` chooses all features whose p-value is below a threshold, thus controlling the false positive rate of selection.
 
-By default, the selection method is `KBest`, the default number of top features is 50. User can use
-`setNumTopFeatures`, `setPercentile` and `setAlpha` to set different selection methods.
+By default, the selection method is `numTopFeatures`, with the default number of top features set to 50. User can use
+`setNumTopFeatures`, `setPercentile`, `fpr` to set different selection methods.
```
Comment (mpjlu) [inline docs/mllib-feature-extraction.md]: ditto

File: mllib/src/main/scala/org/apache/spark/ml/feature/ChiSqSelector.scala [hunk 5]
```diff
@@ -44,67 +44,78 @@ private[feature] trait ChiSqSelectorParams extends Params
...
 
   /**
    * Percentile of features that selector will select, ordered by statistics value descending.
    * Only applicable when selectorType = "percentile".
    * Default value is 0.1.
+   * @group param
    */
+  @Since("2.1.0")
   final val percentile = new DoubleParam(this, "percentile",
     "Percentile of features that selector will select, ordered by statistics value descending.",
```
Comment (mpjlu) [inline mllib/src/main/scala/org/apache/spark/ml/feature/ChiSqSelector.scala]: ordered by pValue ascending.

File: mllib/src/main/scala/org/apache/spark/ml/feature/ChiSqSelector.scala [hunk 6]
```diff
@@ -44,67 +44,78 @@ private[feature] trait ChiSqSelectorParams extends Params
...
 /**
  * Chi-Squared feature selection, which selects categorical features to use for predicting a
  * categorical label.
- * The selector supports three selection methods: `kbest`, `percentile` and `fpr`.
- * `kbest` chooses the `k` top features according to a chi-squared test.
- * `percentile` is similar but chooses a fraction of all features instead of a fixed number.
- * `fpr` chooses all features whose false positive rate meets some threshold.
- * By default, the selection method is `kbest`, the default number of top features is 50.
+ * The selector supports different selection methods: `numTopFeatures`, `percentile`, `fpr`.
+ *  - `numTopFeatures` chooses the `k` top features according to a chi-squared test.
```
Comment (mpjlu) [inline mllib/src/main/scala/org/apache/spark/ml/feature/ChiSqSelector.scala]: ditto

File: mllib/src/main/scala/org/apache/spark/mllib/feature/ChiSqSelector.scala [hunk 7]
```diff
@@ -171,18 +171,19 @@ object ChiSqSelectorModel extends Loader[ChiSqSelectorModel] {
 
 /**
  * Creates a ChiSquared feature selector.
- * The selector supports three selection methods: `kbest`, `percentile` and `fpr`.
- * `kbest` chooses the `k` top features according to a chi-squared test.
- * `percentile` is similar but chooses a fraction of all features instead of a fixed number.
- * `fpr` chooses all features whose false positive rate meets some threshold.
- * By default, the selection method is `kbest`, the default number of top features is 50.
+ * The selector supports different selection methods: `numTopFeatures`, `percentile`, `fpr`.
+ *  - `numTopFeatures` chooses the `k` top features according to a chi-squared test.
```
Comment (mpjlu) [inline mllib/src/main/scala/org/apache/spark/mllib/feature/ChiSqSelector.scala]: ditto

File: mllib/src/test/scala/org/apache/spark/mllib/feature/ChiSqSelectorSuite.scala [hunk 8]
```diff
@@ -54,33 +54,34 @@ class ChiSqSelectorSuite extends SparkFunSuite with MLlibTestSparkContext {
...
     val filteredData = labeledDiscreteData.map { lp =>
       LabeledPoint(lp.label, model.transform(lp.features))
-    }.collect().toSet
-    assert(filteredData == preFilteredData)
+    }.collect().toSeq
+    assert(filteredData === preFilteredData)
   }
 
-  test("ChiSqSelector by FPR transform test (sparse & dense vector)") {
+  test("ChiSqSelector by percentile transform test (sparse & dense vector)") {
```
Comment (mpjlu) [inline mllib/src/test/scala/org/apache/spark/mllib/feature/ChiSqSelectorSuite.scala]: should be FPR

File: docs/ml-features.md (hunk 2 above)
Comment (yanboliang) [inline docs/ml-features.md]: Yeah, we use ```setSelectorType``` to switch between different method. Actually the document is incorrect before this PR.

File: mllib/src/main/scala/org/apache/spark/ml/feature/ChiSqSelector.scala [hunk 9]
```diff
@@ -44,67 +44,78 @@ private[feature] trait ChiSqSelectorParams extends Params
   /**
    * Number of features that selector will select (ordered by statistic value descending). If the
```
Comment (yanboliang) [inline mllib/src/main/scala/org/apache/spark/ml/feature/ChiSqSelector.scala]: ```ordered by pValue ascending```

**Comment by jkbradley** [comment]
Updated---thanks!

**Comment by jkbradley** [comment]
Any more comments, or shall I merge this?

**Comment by jkbradley** [comment]
test this please

**Comment by jkbradley** [comment]
I'll go ahead and merge this, but please comment if it needs any follow-ups.

Merging with master

Thanks for the review!

**Pull request closed without merging**